File: game-admin/src/main/java/com/dkm/updateImage/services/impl/PictureServiceImpl.java
package com.dkm.updateImage.services.impl;

import com.dkm.constanct.CodeType;
import com.dkm.data.Result;
import com.dkm.entity.vo.FileVo;
import com.dkm.exception.ApplicationException;
import com.dkm.feign.AnotherFileFeignClient;
import com.dkm.updateImage.entities.ImageUpdateResultVo;
import com.dkm.updateImage.services.IPictureService;
import org.jetbrains.annotations.NotNull;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import javax.annotation.Resource;

@Service
public class PictureServiceImpl implements IPictureService {
   private final Logger logger = LoggerFactory.getLogger(this.getClass());
   @Resource
   private AnotherFileFeignClient anotherFileFeignClient;

   @NotNull
   @Override
   public ImageUpdateResultVo update(@NotNull MultipartFile multipartFile, @NotNull String token) {
      logger.info(multipartFile.getName());
      logger.info(String.valueOf(multipartFile.getSize()));
      logger.info(multipartFile.getOriginalFilename());

      //通过调用文件服务
      //生成自己名片的二维码并上传服务器
      Result<FileVo> var7 = anotherFileFeignClient.storeFile(token, multipartFile);
      FileVo var8 = var7.getData();
      if (var8 != null) {
         return new ImageUpdateResultVo(var8.getFileUrl());
      } else {
         throw (new ApplicationException(CodeType.FEIGN_CONNECT_ERROR, "请求文件上传失败！"));
      }
   }
}

File: game-admin/src/main/java/com/dkm/updateImage/services/impl/PictureServiceImpl.kt
package com.dkm.updateImage.services.impl

import com.dkm.constanct.CodeType
import com.dkm.exception.ApplicationException
import com.dkm.feign.AnotherFileFeignClient
import com.dkm.updateImage.entities.ImageUpdateResultVo
import com.dkm.updateImage.services.IPictureService
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.web.multipart.MultipartFile
import javax.annotation.Resource

@Service
class PictureServiceImpl : IPictureService {
    @Resource
    private lateinit var anotherFileFeignClient: AnotherFileFeignClient

    private val logger = LoggerFactory.getLogger(javaClass)
    override fun update(multipartFile: MultipartFile, token: String) = kotlin.run {
        logger.info(multipartFile.name)
        logger.info(multipartFile.size.toString())
        logger.info(multipartFile.originalFilename)
        val storeFile = anotherFileFeignClient.storeFile(token,multipartFile).data
                ?: throw ApplicationException(CodeType.FEIGN_CONNECT_ERROR, "请求文件上传失败！")
        ImageUpdateResultVo(storeFile.fileUrl)
    }

}
